File: src/main/Java/boj_step4/boj1546.java
import java.io.*;
import java.util.Arrays;

public class boj1546 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int count = Integer.parseInt(reader.readLine());
        double[] score = Arrays.stream(reader.readLine().split(" ")).mapToDouble(Double::parseDouble).toArray();
        double maxScore = Arrays.stream(score).max().getAsDouble();
        int index = 0;
        for (double it: score) score[index++] = it / maxScore * 100;
        writer.append(String.valueOf(Arrays.stream(score).average().getAsDouble()));
        writer.close();
    }
}

File: src/main/Kotlin/others/boj1546.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine()!!.toInt()
    var scores = r.readLine()!!.split(" ").map { it.toInt() }
    val max = scores.max()
    var avg = .0
    for (i in 0 until scores.size) avg += scores[i] / max.toDouble()
    w.write((avg / n.toDouble() * 100).toString())
    w.close()
}